taskcluster/src/transforms/__init__.py

```diff
@@ -105,3 +106,20 @@ def parameterize_mounts(config, jobs):
                             rust_version=rust_version,
                             node_version=node_version)
         yield job
+
+
+@transforms.add
+def docker_worker_chunk(config, jobs):
+    total_chunks = 5
+    for job in jobs:
+        for chunk in range(1, total_chunks + 1):
+            c_task = copy.deepcopy(job)
+            c_task["name"] += f"-{chunk}"
+            c_task["description"] += f" #{chunk}"
+            c_task["run"]["command"] += f" --this-chunk {chunk} --total-chunks {total_chunks}"
+
+            # include release-publish dependency in the case of a release
+            if config.params["tasks_for"] == "github-push":
```

Comment:
I guess this is not enough to determine if this is a release or not.

https://github.com/taskcluster/taskcluster/blob/6185123416f71712fc63eb69a3505bbfac4862a3/taskcluster/src/target_tasks.py#L7-L11

Or this will also be controlled by the logic above `only-on` and doesn't need to be added here. Hm.. @ahal? :)

Reply:
Yeah, you might want to create a new parameter called `release` or something that does this early on in the generation, and then that way you can re-use it throughout.

Alternatively you could fake the `tasks_for` value in your `.taskcluster.yml` to be `github-release` when a tag push is detected... but that feels hacky to me.